bchrpc/bchrpc.proto

```diff
@@ -472,6 +523,73 @@ message SubscribeBlocksRequest {
     bool serialize_block = 3;
 }
 
+message GetTokenMetadataRequest {
+    repeated bytes token_ids = 1;
+}
+
+message GetTokenMetadataResponse {
+    repeated TokenMetadata token_metadata = 1;
+}
+
+message GetParsedSlpScriptRequest {
+    bytes slp_opreturn_script = 1;
+}
+
+message GetParsedSlpScriptResponse {
+    string parsing_error = 1;
+    bytes token_id = 2;
+    SlpVersionType type = 3;
+    oneof slp_metadata {
+        SlpV1GenesisMetadata v1_genesis = 4;    // NFT1 Group also uses this
+        SlpV1MintMetadata v1_mint = 5;          // NFT1 Group also uses this
+        SlpV1SendMetadata v1_send = 6;          // NFT1 Group also uses this
+        SlpNft1ChildGenesisMetadata nft1_child_genesis = 7;
+        SlpNft1ChildSendMetadata nft1_child_send = 8;
+    }
+}
+
+message GetTrustedSlpValidationRequest {
+    message Query {
+        bytes prev_out_hash = 1;
+        uint32 prev_out_vout = 2;
+    }
+    repeated Query queries = 1;
+}
+
+message GetTrustedSlpValidationResponse {
+    message ValidityResult {
+        bytes prev_out_hash = 1;
+        uint32 prev_out_vout = 2;
+        bytes token_id = 3;
+        SlpVersionType slp_version = 4;
+        oneof validity_result_type {
+            uint64 v1_token_amount = 5;
+            bool v1_mint_baton = 6;
+        }
+        bytes slp_txn_opreturn = 7;
+
+        message FunctionarySignature {
+            bytes signature = 1;
+            bytes message = 2;
+        }
+        FunctionarySignature functionary_sig = 8;
+    }
+
+    // allows server to respond to a request with multiple
```

Comment:
NIT: incomplete sentence.

Reply:
Comment still not complete.

Reply:
I will remove the comment, as this is obvious.

Reply:
done